### PR TITLE
Add new unauthenticated route for reporting stats API

### PR DESCRIPTION
**Pull request by abbashus**
*Issue #, if available:*

*Description of changes:*
This change introduces new unauthenticated route for Kibana reporting stats API **api/reporting/stats**.
The API exposes operational and business metrics for Kibana Reporting plugin. 
See https://github.com/opendistro-for-elasticsearch/kibana-reports/pull/277

Sample response:
```
  {
  "report": {  
    "create": {  # POST /api/reporting/generateReport
      "count": 0,
      "system_error": 0,
      "user_error": 3,
      "total": 0
    },
    "create_from_definition": {  # POST /api/reporting/gemerateReport/{reportDefinitionId}
      "count": 1,
      "system_error": 0,
      "user_error": 0,
      "total": 1
    },
    "download": {  # GET /api/reporting/generateReport/{reportId}
      "count": 5,
      "system_error": 0,
      "user_error": 0,
      "total": 5
    },
    "list": {  # GET /api/reporting/reports
      "count": 7,
      "system_error": 0,
      "user_error": 0,
      "total": 7
    },
    "info": {  # GET /api/reporting/reports/{reportId}
      "count": 1,
      "system_error": 0,
      "user_error": 0,
      "total": 1
    }
  },
  "report_definition": {
    "create": {  # POST /api/reporting/reportDefinition
      "count": 0,
      "system_error": 0,
      "user_error": 0,
      "total": 0
    },
    "list": {  # GET /api/reporting/reportDefinitions
      "count": 7,
      "system_error": 0,
      "user_error": 0,
      "total": 7
    },
    "info": {  # GET /api/reporting/reportDefinition/{reportDefinitionId}
      "count": 3,
      "system_error": 0,
      "user_error": 0,
      "total": 3
    },
    "update": {   # PUT /api/reporting/reportDefinition/{reportDefinitionId}
      "count": 0,
      "system_error": 0,
      "user_error": 0,
      "total": 0
    },
    "delete": {   # DELETE /api/reporting/reportDefinition/{reportDefinitionId}
      "count": 1,
      "system_error": 0,
      "user_error": 0,
      "total": 1
    }
  },
  "report_source": {   # GET /api/reporting/reportSource/{reportSourceType}
    "list": {
      "count": 3,
      "system_error": 0,
      "user_error": 0,
      "total": 3
    }
  },
  "dashboard": {
    "pdf": {
      "download": {
        "count": 1,
        "total": 1
      }
    },
    "png": {
      "download": {
        "count": 1,
        "total": 1
      }
    }
  },
  "visualization": {
    "pdf": {
      "download": {
        "count": 0,
        "total": 0
      }
    },
    "png": {
      "download": {
        "count": 0,
        "total": 0
      }
    }
  },
  "saved_search": {
    "csv": {
      "download": {
        "count": 4,
        "total": 4
      }
    }
  }
}
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by yuxitang-amzn** [comment]
Could you state in the description what is this API for? Is it for reporting kibana plugin?

**Comment by abbashus** [comment]
@yuxitang-amzn Added API related info in the PR description.

**Review by yuxitang-amzn — APPROVED** [review]
LGTM

File: server/index.ts [hunk 1]
```diff
@@ -69,13 +69,13 @@ export const configSchema = schema.object({
...
     }),
     anonymous_auth_enabled: schema.boolean({ defaultValue: false }),
-    unauthenticated_routes: schema.arrayOf(schema.string(), { defaultValue: ['/api/status'] }),
+    unauthenticated_routes: schema.arrayOf(schema.string(), { defaultValue: ['/api/status', '/api/reporting/stats'] }),
     forbidden_usernames: schema.arrayOf(schema.string(), { defaultValue: [] }),
     logout_url: schema.string({ defaultValue: '' }),
   }),
   basicauth: schema.object({
     enabled: schema.boolean({ defaultValue: true }),
-    unauthenticated_routes: schema.arrayOf(schema.string(), { defaultValue: ['/api/status'] }),
```
Comment (zengyan-amazon) [inline server/index.ts:78]: since you have whitelisted the API in `auth.unauthenticated_routes` you don't necessarily add it to basicauth config here

Comment (abbashus) [reply server/index.ts:78]: Okay, I will remove this.

Comment (abbashus) [reply server/index.ts]: Removed.

**Comment by abbashus** [comment]
How can I get build artifact for this PR. I don't see any GH action invoked.

**Review by zengyan-amazon — APPROVED** [review]

**Pull request merged**